packages/components/src/tools-panel/tools-panel/hook.ts

```diff
@@ -62,7 +71,18 @@ export function useToolsPanel(
 	const [ panelItems, setPanelItems ] = useState< ToolsPanelItem[] >( [] );
 
 	const registerPanelItem = ( item: ToolsPanelItem ) => {
-		setPanelItems( ( items ) => [ ...items, item ] );
+		setPanelItems( ( items ) => {
+			// If an item with this label is already registered, remove it first.
+			// This can happen when an item is moved between the default and optional
+			// groups.
+			const existingIndex = items.findIndex(
+				( oldItem ) => oldItem.label === item.label
+			);
+			if ( existingIndex !== -1 ) {
+				items.splice( existingIndex, 1 );
```

Comment:
This mutates `items`. I'm not sure if it will resolve anything, but I was wondering if it's worth using the `newItems = [ ...items ]` approach at the top of this callback, like you've done in `deRegisterPanelItem`?

Reply:
Good catch! I don't think that will cause the described issue, but definitely should be fixed. Done 👍